src/dataloader.py

```diff
@@ -21,6 +21,10 @@ def download_data(self):
             url = 'https://drive.google.com/uc?id=1jZP1bfGTjoz-f-VzHPmzH6YU6ZaO_JQa'
             output = os.path.join(PROJECT_PATH, 'data', self.dataset_name + ".csv")
             gdown.download(url, output, quiet=False)
+        if not os.path.exists(os.path.join(PROJECT_PATH, 'data', 'existing_stations' + ".json")):
```

Comment:
In theory, `gdown` can handle link to shared folders - could you take a look at it? If it does, then collapse all downloads into one block.